### DISCUSSION TITLE
Patch own installation of package?

### DISCUSSION THREAD

**Discussion by ghost** [Homebrew/discussions#5403] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.3.1
ORIGIN: https://github.com/Homebrew/brew
HEAD: e0bc557e7b991cb23583679e1cf1c8a92b793aeb
Last commit: 6 days ago
Core tap JSON: 25 May 21:25 UTC
Core cask tap JSON: 25 May 21:25 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: Eleanors-Laptop.local
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 3.3.1 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.1/bin/ruby
CPU: octa-core 64-bit arm_ibiza
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.6.0 => /usr/bin/curl
macOS: 14.5-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

The site has docs for patching a formula, but not for patching a package between download and installation. My use case is there's a package written in C, which exposes compile-time options via flags in a header file; I want some options set differently from how they are by default, which involves patching the file and compiling, but I also want to leverage Brew's build and upgrade system. Is that possible?

**Comment by SMillerDev** [comment]
No, you can either use brews configuration or use `brew extract` to make a copy and maintain it for yourself.

**Reply by ghost** [reply]
This sucks. No options in `core`, no local patches...there really should be some middle ground between "exactly one way" and "do everything manually". I'm perfectly willing to rewrite the patch file when it desynchronises and own any failures, but a soft fork is _way_ overkill.

**Comment by gromgit** [comment]
# WARNING: UNTESTED

I think you can riff off the very docs you were reading (I assume you're referring to [this one](https://docs.brew.sh/How-To-Open-a-Homebrew-Pull-Request)): create a branch exactly as if you were going to open a PR (say, `my_custom`), except this branch is now long-lived. IIRC, `brew update` will fetch directly to the Homebrew core `master` branch, so you keep your local core tap on `my_custom` and `git -C $(brew --repo homebrew/core) rebase master` after every `brew update`.

I'm pretty sure this would never be supported by Homebrew maintainers, but as long as you solve your own issues, nobody has to know...
